crates/orchestrator/src/lib.rs

```diff
@@ -416,11 +434,26 @@ where
     }
 
     // Assumes nodes do not post 'ready' twice
-    // TODO ED Add a map to verify which nodes have posted they're ready
-    fn post_ready(&mut self) -> Result<(), ServerError> {
+    fn post_ready(&mut self, peer_config: &PeerConfig<KEY>) -> Result<(), ServerError> {
+        // If we have not disabled registration verification.
+        if self.config.enable_registration_verification {
```

Comment:
I think we can skip the `enable_registration_verification` check here too -- nodes shouldn't be able to post ready unless they're on the `known_nodes_with_stake` list (this is the one we update internally when they register with us)